derive/src/pyclass.rs

```diff
@@ -20,6 +20,10 @@ enum ClassItem {
         py_name: String,
         setter: bool,
     },
+    Slot {
```

Comment:
What is the function of this new concept of slot? Could you try to explain a bit about it?

Reply:
It's something CPython uses to associate functions with a type without exposing them to Python code, usually for optimization. There's a lot more slots in CPython for operators, descriptors, getattr{ibute}, etc, but right now I only added `new` because the way that it works in conjunction with type_call and the [type_new wrapper](https://github.com/python/cpython/blob/master/Objects/typeobject.c#L6032) is very complicated, and it needed a separation of functionality from the type-checking and stuff.